### PR TITLE
Implemented authentication middleware for admin

### PR DESCRIPTION
**Pull request by fob413**

### CODE REVIEW

File: src/middleware/api/adminAuth.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        const token = request.headers.authorization;
+
+        if (!token){
+            return response.status(401).json({
+                status: 'error',
+                message: 'No token provided'
+            });
+        }
+
+        const decodedToken = jwt.verify(token, process.env.SECREKEY)
```
Comment (fob413) [inline src/middleware/api/adminAuth.js]: You misspelled your secretkey as `SECREKEY`. It should be `SECRETKEY`

File: src/middleware/api/adminAuth.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        if (!decodedToken.id){
+            return response.status(401).json({
+                status: 'error',
+                message: 'Invalid token'
+            })
+        }
+
+        const id = decodedToken.id;
```
Comment (fob413) [inline src/middleware/api/adminAuth.js]: Use destructuring assignment to get the `id` here to make it look sexy 😉

File: src/middleware/api/adminAuth.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        const admin = await getAdminById(id);
+        if (!admin) {
+            return response.status(401).json({
+                status: 'error',
+                message: 'Admin does not exist'
+            })
+        }else{
+            request.admin = admin;
```
Comment (fob413) [inline src/middleware/api/adminAuth.js:38]: Nicely Done

File: src/middleware/api/adminAuth.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
+const jwt = require('jsonwebtoken');
+const db = require('../../database/db');
+require('dotenv').config();
+
+const getAdminById = async (id) => {
+    const query = 'SLECT * FROM admin WHERE id = $1'
```
Comment (fob413) [inline src/middleware/api/adminAuth.js]: Typo Error here. It should be `SELECT` not `SLECT`

**Review by okikiola-jkt — APPROVED** [review]

**Pull request merged**